gnark/gnark-jni/gnark-eip-196.go

```diff
@@ -288,63 +259,45 @@ func safeUnmarshalEIP196(g1 *bn254.G1Affine, input []byte, offset int) error {
 		pointBytes = input[offset : offset+64]
 	}
 
-	if !checkInFieldEIP196(pointBytes[0:32]) {
-		return ErrPointNotInFieldEIP196
+	if err := g1.X.SetBytesCanonical(pointBytes[0:32]); err != nil {
+		return errCodePointNotInFieldEIP196
 	}
-
-	err := g1.X.SetBytesCanonical(pointBytes[0:32])
-
-	if err == nil {
-
-		if !checkInFieldEIP196(pointBytes[32:64]) {
-			return ErrPointNotInFieldEIP196
-		}
-		err := g1.Y.SetBytesCanonical(pointBytes[32:64])
-		if err == nil {
-			if !g1.IsOnCurve() {
-				return ErrPointOnCurveCheckFailedEIP196
-			}
-			return nil
-		}
+	if err := g1.Y.SetBytesCanonical(pointBytes[32:64]); err != nil {
+		return errCodePointNotInFieldEIP196
+	}
+	if !g1.IsOnCurve() {
+		return errCodePointOnCurveCheckFailedEIP196
```

Comment:
fwiw, if we are trying to eke out additional performance, the isOnCurve() checks are duplicated by SetBytesCanonical since gnark-crypto 0.17.0.  Doing duplicate checks were kept out of an abundance of caution.  see https://github.com/hyperledger/besu-native/pull/262#issuecomment-2831099107 for context

it is worth at least testing without the duplicate isOnCurve check to determine if the impact is negligible enough to keep for "visibility" reasons within the code.

Reply:
Yep, that's on my list but would rather keep this as incremental as possible - would save that for another PR.